docker.go

```diff
@@ -684,15 +722,34 @@ func (c *DockerContainer) StartLogProducer(ctx context.Context) error {
 				b := make([]byte, count)
 				_, err = io.ReadFull(r, b)
 				if err != nil {
-					// TODO: add-logger: use logger to log out this error
-					_, _ = fmt.Fprintf(os.Stderr, "error occurred reading log with known length %s", err.Error())
-					if errors.Is(err, context.DeadlineExceeded) {
-						// Probably safe to continue here
+					select {
+					case <-ctx.Done():
+						return
+					default:
+						c.logger.Printf("error occurred reading log with known length %s", err.Error())
+					}
+
+					if errors.Is(err, context.DeadlineExceeded) || errors.Is(err, context.Canceled) {
```

Comment:
The `err` here is from the `io.ReadFull()` so we shouldn't really be concerned about the context cancellation here.

Reply:
Do you think we should not override the outer err and still check for the outer context cancellation?

Reply:
It's not overridden, is it? Between line 723 and here - 732 - nothing writes to `err`.

Reply:
Oh you are right, the initial err comes from another read: `_, err := io.ReadFull(r, h)` Sorry, my bad